gateway.py

```diff
@@ -685,10 +676,19 @@ def update_listing(http: urllib3.PoolManager, auth_token, listing_id, title, pri
             "status": status,
             "charity": charity
         })
-    if result.status == 200:
-        return make_ok_response()
-    elif result.status == 400:
+    if update_result.status == 400:
         return make_invalid_request_response("Invalid request")
+
+    if status == 'SOLD' and buyer_username is not None:
+        sale_result = execute_data_post(
+            http, f"/create_sale", {
```

Comment:
there's no lookup to check if the buyer username is valid, so if an invalid buyer username is provided then the listing will be marked as sold but a record won't be created in the `Sales` table